src/main/java/com/griddynamics/jagger/jenkins/jaas/plugin/JaggerTestExecutionBuilder.java

```diff
@@ -219,13 +227,54 @@ private TestExecutionEntity waitTestExecutionFinished(PrintStream logger, Long e
         return execution;
     }
 
-    private void publishReportLink(PrintStream logger, TestExecutionEntity executionFinished) {
-        if (executionFinished.getSessionId() == null)
+    private void publishReportLink(PrintStream logger, AbstractBuild<?, ?> build, TestExecutionEntity executionFinished) {
+        if (executionFinished.getSessionId() == null) {
             logger.println("sessionId is unavailable. Can’t publish link to the test report.");
+            build.setResult(Result.UNSTABLE);
+        }
         else
             logger.println("Test execution report can be found by the link " + evaluatedJaasEndpoint + "/report?sessionId=" + executionFinished.getSessionId());
     }
 
+    private void checkDecision(PrintStream logger, AbstractBuild<?, ?> build, String sessionId, RestTemplate restTemplate) throws AbortException {
+        try {
+            logger.println();
+            logger.println(format("Checking decision for test session with id=%s ... ", sessionId));
+            String decisionURL = evaluatedJaasEndpoint + "/db/sessions/" + sessionId + "/decision";
+            RequestEntity<?> requestEntity = RequestEntity.get(new URI(decisionURL)).build();
+            ResponseEntity<DecisionPerSessionDto> responseEntity = restTemplate.exchange(requestEntity, DecisionPerSessionDto.class);
+            String decisionJson = new ObjectMapper().writerWithDefaultPrettyPrinter().writeValueAsString(responseEntity.getBody());
+            Decision decision = responseEntity.getBody().getDecision();
+            logger.println(format("Decision for session with id=%s is %s", sessionId, decision));
+            if (decision == Decision.OK) {
+                logger.println("Full session decision can be found here: " + decisionURL);
+            } else {
+                logger.println("Full session decision:\n" + decisionJson);
+            }
+            if (decision == Decision.ERROR || decision == Decision.FATAL) {
+                build.setResult(Result.FAILURE);
+            } else if (decision == Decision.WARNING) {
+                build.setResult(Result.UNSTABLE);
+            }
+        } catch (URISyntaxException e) {
+            logger.println();
+            throw new AbortException("Invalid JaaS endpoint URL: " + e.getMessage());
+        } catch (RestClientException ex) {
+            build.setResult(Result.UNSTABLE);
+            logger.println();
+            if ((ex instanceof HttpClientErrorException) && ((HttpClientErrorException) ex).getRawStatusCode() == 404) {
+                logger.println(format("Error occurred while checking decision of Test session with id=%s: " +
+                        "Endpoint %s/db/sessions/%s/decision is not found. Seems that property 'jaas.hide.db.access.via.api' is set to false.",
```

Comment:
propose not to mention jaas properties here. we will change them in jaas and will forget to update them here